src/Path.hs

```diff
@@ -330,6 +332,41 @@ dirname (Path l) = Path (last (FilePath.splitPath l))
 fileExtension :: Path b File -> String
 fileExtension = FilePath.takeExtension . toFilePath
 
+-- | Add extension to given file path. Throws if the
+-- resulting filename does not parse.
+--
+-- >>> addFileExtension "symbols" $(mkRelFile "Data.List")
+-- "Data.List.symbols"
+-- >>> addFileExtension "evil/" $(mkRelFile "Data.List")
+-- *** Exception: InvalidRelFile "Data.List.evil/"
+--
+-- @since 0.6.1
+addFileExtension :: MonadThrow m
```

Comment:
@joehillen I did it similar to `setFileExtension`. I thought consistency is better.